### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by Martoon-00**

### CODE REVIEW

File: src/Node/Internal.hs [hunk 1]
```diff
@@ -354,24 +463,26 @@ nodeDispatcher endpoint nodeState handlerIn handlerInOut =
...
                   case outcome of
                       Nothing -> pure Nothing
-                      Just (tid, ChannelIn chan) -> do
+                      Just (promise, ChannelIn chan) -> do
                           Channel.writeChannel chan (Just (BS.concat (LBS.toChunks ws')))
-                          pure . Just $ ConnectionReceiving tid chan
+                          pure . Just $ ConnectionReceiving promise (ChannelIn chan)
 
                 | otherwise ->
                     throw (ProtocolError $ "unexpected control header " ++ show w)
```
Comment (avieth) [inline src/Node/Internal.hs]: Log don't throw.

File: src/Node/Internal.hs [hunk 2]
```diff
@@ -383,31 +494,48 @@ nodeDispatcher endpoint nodeState handlerIn handlerInOut =
...
-                  Just (ConnectionNewChunks peer@(NodeId _) chunks0) -> do
-                      mConnState <- handleFirstChunks connid peer (chunks0 ++ chunks) state
-                      loop (maybe state (\cs -> Map.insert connid cs state) mConnState)
+                  Just (peer, ConnectionNew) ->
+                      loop $ state {
+                            dispatcherConnections = Map.insert connid (peer, ConnectionNewChunks chunks) (dispatcherConnections state)
+                          }
+
+                  Just (peer, ConnectionNewChunks chunks0) -> do
+                      emConnState <- try $ handleFirstChunks connid peer (chunks0 ++ chunks) state
```
Comment (avieth) [inline src/Node/Internal.hs]: Skeptical of using `try` here. Exceptions in `handleFirstChunks` should be re-thrown I think.

File: src/Node/Internal.hs [hunk 3]
```diff
@@ -417,71 +545,147 @@ nodeDispatcher endpoint nodeState handlerIn handlerInOut =
...
+                            dispatcherConnections = Map.insert connid (peer, ConnectionClosed promise) (dispatcherConnections state)
+                          }
 
                   -- Small message
-                  Just (ConnectionNewChunks peer chunks0) -> do
-                      mConnState <- handleFirstChunks connid peer chunks0 state
-                      case mConnState of
-                          Just (ConnectionReceiving tid chan) -> do
+                  Just (peer, ConnectionNewChunks chunks0) -> do
+                      emConnState <- try $ handleFirstChunks connid peer chunks0 state
```
Comment (avieth) [inline src/Node/Internal.hs]: See previous comment about use of `try` on `handleFirstChunks`.

File: src/Node/Internal.hs (hunk 2 above)
Comment (Martoon-00) [inline src/Node/Internal.hs]: AFAIU if I rethrow, then dispatcher loop would stop. Is it okay?

Comment (avieth) [inline src/Node/Internal.hs]: Yes I think this is what we want. If there's an exception then we really do want to know about it.

**Comment by Martoon-00** [comment]
Okay, I'm going to apply your notes, thanks for review!

**Pull request closed without merging**